core/src/main/scala/kafka/server/KafkaRequestHandler.scala

```diff
@@ -108,11 +108,10 @@ class KafkaRequestHandler(id: Int,
           completeShutdown()
           return
 
-        case request: RequestChannel.CallbackRequest =>
+        case callback: RequestChannel.CallbackRequest =>
           try {
-            request.originalRequest.callbackRequestDequeTimeNanos = Some(time.nanoseconds())
-            request.fun() 
-            request.originalRequest.callbackRequestCompleteTimeNanos = Some(time.nanoseconds())
+            callback.originalRequest.callbackRequestDequeTimeNanos = Some(time.nanoseconds())
+            callback.fun()
```

Comment:
I think it's still good to keep the request.originalRequest.callbackRequestCompleteTimeNanos = Some(time.nanoseconds()) here as well.  Looking at the code, request could end in many different ways, some could happen before callback completion, some may happen after callback completion.  There are some error cases, like connection disconnects that may not go through the success path.